transformer_engine/jax/praxis/transformer.py

```diff
@@ -59,30 +62,114 @@ def __call__(self, q_seqlen: JTensor, k_seqlen: JTensor, bidirectional: bool = T
         return self.relative_position_bias(q_seqlen, k_seqlen, bidirectional)
 
 
+class DotProductAttention(TransformerEngineBaseLayer):
+    """DotProductAttention"""
+
+    head_dim: int = 64
+    num_attention_heads: int = 16
+    num_gqa_groups: int | None = None
+    attention_dropout: float = 0.
```

Comment:
Discussion: This is the default in TE's DPA API in PyTorch and JAX MHA API, but I'm not sure if this is the best default? Would also need fixing in the other APIs.

Reply:
I will remove the default of `head_dim` and `num_attetnion_heads` here and also `MultiHeadAttention`'s [default](https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/jax/flax/transformer.py#L353-L354) for those two parameters in `TE-JAX-FLAX`. I checked that there is no default for those two parameters in [TE-PyT](https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/attention.py#L2315-L2316) and [TE-JAX-FLAX](https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/jax/flax/transformer.py#L353-L354) DPA/MHA API.

Reply:
Fix in 8666d48a3da5f7220514646d8eb0d8c48c068bc9. Praxis layers are not allowed the no-default members, so I give the default value = 0 and check if it is set in the `setup()`, which is the same way used in the praxis repo ([set zero](https://github.com/google/praxis/blob/main/praxis/layers/attentions.py#L1149) and [check](https://github.com/google/praxis/blob/main/praxis/layers/attentions.py#L1234)).